sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java

```diff
@@ -1108,7 +1156,8 @@ private void runTasksInPendingAcquisitionQueue() {
 
             task.acquired(true);
             this.acquire(task.promise);
-        }
+
+        } while (this.availableChannels.size() > 0);
```

Comment:
Should be a while block to ensure we only acquire channel if at least one is available ?

Reply:
thank you